desktop-widgets/tab-widgets/maintab.cpp

```diff
@@ -210,21 +210,6 @@ void MainTab::enableEdition(EditMode newEditMode)
 {
 	if (((newEditMode == DIVE || newEditMode == NONE) && current_dive == NULL) || editMode != NONE)
 		return;
-	if ((newEditMode == DIVE || newEditMode == NONE) &&
-	    current_dive->dc.model &&
-	    strcmp(current_dive->dc.model, "manually added dive") == 0) {
-		// editCurrentDive will call enableEdition with newEditMode == MANUALLY_ADDED_DIVE
-		// so exit this function here after editCurrentDive() returns
-
-
-
-		// FIXME : can we get rid of this recursive crap?
-
-
-
-		MainWindow::instance()->editCurrentDive();
-		return;
-	}
```

Comment:
To my understanding, the previous commits made this dead code:
The only caller of `enableEdition()` is `MainWindow::editCurrentDive()` and that *always* calls `enableEdition()` with `mode == MANUALLY_ADDED_DIVE` for manually added dives. So we can never reach this if-branch (note that it would lead to an infinite recursion).

This was only necessary when `enableEdition()` was called from other parts of the code.